src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -168,7 +168,7 @@ public static bool TryGetRecordPrimaryConstructor(this INamedTypeSymbol typeSymb
                 // of the extension method and make sure the change is applicable to all these usages.
 
                 primaryConstructor = typeSymbol.InstanceConstructors.FirstOrDefault(
-                    c => c.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax() is RecordDeclarationSyntax);
+                    c => c.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax() is RecordDeclarationSyntax or ClassDeclarationSyntax or StructDeclarationSyntax);
```

Comment:
```suggestion
                    c => c.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax() is TypeDeclarationSyntax);
```

```suggestion
                    c => c.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax() is not ConstructorDeclarationSyntax);
```

Though, actually, this logic slightly worries me.  Won't this return 'true' for the implicitly created constructor for a struct that does *not* have an actual primary constructor?  Maybe we should say that the syntax has to both be the DeclSyntax *And* the decl syntax has a primary constructor?

Reply:
I prefer to follow the pattern established for records. ```TypeDeclarationSyntax``` would include interfaces and this is not my intention.